cloud-controller-manager/do/cloud.go

```diff
@@ -109,7 +112,13 @@ func newCloud() (cloudprovider.Interface, error) {
 
 	clusterID := os.Getenv(doClusterIDEnv)
 	clusterVPCID := os.Getenv(doClusterVPCIDEnv)
-	resources := newResources(clusterID, clusterVPCID, doClient)
+	firewallName := os.Getenv(publicAccessFirewallNameEnv)
+	firewallTags := os.Getenv(publicAccessFirewallTagsEnv)
+	if firewallName != "" && firewallTags == "" {
```

Comment:
I wonder if we should be more explicit here by error out if either firewall-related environment variable is missing. My thinking here is that it could be confusing to users if they specified the tags but unintentionally left out the name, wondering why things are not working. WDYT?

Reply:
Doesn't this alert the user? :

```go
if c.resources.publicAccessFirewallName == "" {
	klog.Info("Nothing to manage since firewall name was not provided")
	return
}
```

https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/d4461caec230d3a78ff68b8d1666da63fa54f964/cloud-controller-manager/do/cloud.go#L163-L166

_or_

Do we prefer to remove the check for the empty firewall name and just error out anytime the firewall name is left empty? I think we should only error out on tags being empty and keep the `klog` statement for when the firewall name is empty. Wdyt @timoreimann ?

Reply:
Now, that I think about it. Maybe it doesn't hurt to return an error in the case where tags are provided and the name is not. Because they are expecting something to happen when they provide at least one of those.

Shouldn't I remove the `klog` statement Since it doesn't look like we would get there if we are failing in `newCloud()`, right? @timoreimann

Reply:
Well, in the case that `firewallName` and `firewallTags` are both `""`, it would reach this code. So maybe let's leave it @timoreimann .

Reply:
Yeah looks good this way. 👍